src/generic.c

```diff
@@ -149,12 +149,15 @@ generic_set_socket_buffer(int fd, int size)
 
 	if (w == 0)
 		return (UPERF_SUCCESS);
-	if (setsockopt(fd, SOL_SOCKET, SO_SNDBUF, (char *)&w, sizeof (w))) {
+	if (setsockopt(fd, SOL_SOCKET, SO_SNDBUF, (char *)&w, sizeof (w))
+		!= 0) {
 		ulog_warn("Cannot set SO_SNDBUF");
+		errno = 0;
```

Comment:
Why do you need to set errno? One should only look at errno once after a failed system call. In other cases, the value is not specified.